x-pack/docs/en/security/auditing/event-types.asciidoc

```diff
@@ -343,15 +343,21 @@ that have been previously described:
                             uses the <<run-as-privilege, run as authorization functionality>>.
   `authentication.type`::   Method used to authenticate the user.
                             Possible values are `REALM`, `API_KEY`, `TOKEN`, `ANONYMOUS` or `INTERNAL`.
-  `api_key.id`         ::   API key ID returned by the <<security-api-create-api-key,create API key>> request.
+  `apikey.id`          ::   API key ID returned by the <<security-api-create-api-key,create API key>> request.
                             This attribute is only provided for authentication using an API key.
-  `api_key.name`       ::   API key name provided in the <<security-api-create-api-key,create API key>> request.
+  `apikey.name`        ::   API key name provided in the <<security-api-create-api-key,create API key>> request.
                             This attribute is only provided for authentication using an API key.
+  `service_token.name` ::   Name of the <<service-accounts,service account>> token.
```

Comment:
Doc building for the `<<service-account>>` link fails because #71729 is still in progress.

Reply:
I wonder whether this should be `authentication.token.name` instead (and maybe a `authentication.token.type` as well)

Reply:
I think this suggestion because:
1. It is more inline with the discussion for future fields such as `api_key`, which can be logged as `authentication.api_key.name` etc.
2. We already have a field `authentication.type` which falls into the same `authentication` grouping nicely and meaninfully.

@albertzaharovits Do you have a preference on this?

Reply:
@ywangd Nesting token details under `authentication.<type>` sounds good to me too. I can't think of drawbacks when mapping and querying, and fewer top level fields is always preferable. In hindsight sounds foolish to have a field that informs that other fields are present, instead of simply using nesting. Thanks for asking.

Reply:
@ywangd Upon further though, I don't think we can make this change in a minor version.
Dots in field names indicate objects and we should assume that audit logs are indexed, so in this case `authentication.token` can be mapped to either `keyword` or `object`.

We have to make the change in v8, where all the authentication types follow the nested object model.

For a long time we've incrementally added pieces to the audit format. I think an overhaul is due. You've already uncovered a couple of incongruities (the most recent that I remember, `realm` should be `realm.name`).

Reply:
Sorry, I don't follow.  
Why is adding `authentication.token.name` (when we already have `authentication.type`) a problem but adding `service_token.name` is OK?

Reply:
My bad, it is not a problem, I got confused between the `type` and `token` fields. Since `authentication.token` does not currently exist, there's no problem.

Reply:
Thank you both. I have [updated](https://github.com/elastic/elasticsearch/pull/72198/commits/8f5b6819aba2e386eec27b42855992da9df110c0) the PR to rename the field. I will also raise another PR for `authentication.token.type` after #73135 is approved and merged.